File: ds&algo/linearSearch.js
const linearSearch = (array, length, number) => {
	for (let i = 0; i < length; i++) {
		if (array[i] === number) {
			console.log(`Found ${number} at index ${i}`);
			return;
		}
	}
	console.log(`${number} not found`);
};

function main() {
	const array = [11, 2, 31, 33, 15, 46, 17, 28, 39, 10];
	const number = 39;
	const length = array.length;
	linearSearch(array, length, number);
}

main();

File: ds&algo@cpp/linearSearch.cpp
#include <iostream>
using namespace std;

int linearSearch(int arr[], int length, int key)
{
    for (int i = 0; i < length; i++)
    {
        if (arr[i] == key)
        {
            return i;
        }
    }
    return -1;
}

int main()
{
    int arr[] = {8, 64, 52, 89, 12};
    int length = 4;
    int key = 89;
    int result = linearSearch(arr, length, key);
    cout << "The " << key << " Found in index " << result << endl;
    return 0;
}